### PR TITLE
Symfony 5.x support

### PR DESCRIPTION
**Pull request by kiettran**
This pull request adds full symfony 5 support. Do note, this pull request does not have symfony 3.4 or symfony 4.x support anymore.

#23

### CODE REVIEW

**Comment by xvilo** [comment]
@kiettran any idea on when you will finish this?

**Comment by kiettran** [comment]
It’s done except two tests are failing because the classes are final and cannot be mocked. The pull request does work within a symfony 5 project.

**Comment by niels-nijens** [comment]
Thanks for the effort (again).

In its current form I'm unable approve this. Symfony 4.4 is supported until the end of 2022 and I'm unwilling to drop support for it when Symfony does not.

Though, I am willing to drop support for Symfony 3.4 to make the upgrade path easier.

A lot is going on in this pull request and it's not just Symfony 5.x support. It should only contain just that.

File: tests/EventListener/JsonRequestBodyValidationSubscriberTest.php [hunk 1]
```diff
@@ -122,7 +122,11 @@ public function testValidateRequestBodySkipsValidationWhenRouteIsNotAvailable()
         $request = new Request();
         $request->headers->set('Content-Type', 'application/json');
 
-        $event = new GetResponseEvent($kernelMock, $request, HttpKernelInterface::MASTER_REQUEST);
+        $event = new RequestEvent(
+            $kernelMock,
+            $request,
+            HttpKernelInterface::MASTER_REQUEST
+        );
```
Comment (niels-nijens) [inline tests/EventListener/JsonRequestBodyValidationSubscriberTest.php:129]: As this is actually a recurring piece of code. We can extract it and make it backward compatible:
```php
    /**
     * Creates a request event.
     *
     * @return RequestEvent|GetResponseEvent
     */
    public function createRequestEvent(Request $request)
    {
        $kernelMock = $this->createMock(HttpKernelInterface::class);

        if (class_exists('Symfony\Component\HttpKernel\Event\GetResponseEvent')) {
            return new GetResponseEvent($kernelMock, $request, HttpKernelInterface::MASTER_REQUEST);
        }

        return new RequestEvent($kernelMock, $request, HttpKernelInterface::MASTER_REQUEST);
    }
```

The same approach can be used in `JsonResponseExceptionSubscriberTest`.

**Comment by niels-nijens** [comment]
I noticed after my review that the change of type hints in the `RouteLoader` might be the only backwards compatibility problem.

For this I tried the following changes:
https://github.com/nijens/openapi-bundle/compare/master...niels-nijens:route-loader-type-hints 

Although it's not the prettiest code, it works like a charm. 😄 

When you rebase to master and possibly fix the conflicts, I can PR it to your branch.

If you need any help getting this ready. Let me know!

**Review by niels-nijens — APPROVED** [review]

**Comment by niels-nijens** [comment]
Thanks!

**Pull request merged**